src/OAuth2.php

```diff
@@ -268,6 +268,13 @@ class OAuth2 implements FetchAuthTokenInterface
      */
     private $additionalClaims;
 
+    /**
+     * Added flag to enable/disable partial consent screen
+     *
+     * @var ?string
+     */
+    private $partConsent;
```

Comment:
The design is not to expose a parameter specifically for partial consent and add it to the auth url. Instead, we should be exposing a property bag with which the user can provide any custom key-value pairs to be added to the auth url. Checkout "Introduce a free-form query parameter-value option" section of go/custom-auth-params. It has a code sample for java as well.

Reply:
Hmm, is this the required implementation? It wouldn't be consistent with how we currently do things if we implemented it this way

Reply:
Well, taking a look at our `buildFullAuthorizationUrl` method, it _already supports_ [additional custom querystring parameters](https://github.com/googleapis/google-auth-library-php/blob/main/src/OAuth2.php#L682) via the `$config` variable. So it's possible that this requirement is already satisfied.

Reply:
Thanks for pointing out the existing implementation for providing custom query parameters.